src/store/mem.rs

```diff
@@ -2,6 +2,7 @@ use std::{collections::HashMap, sync::Arc};
 
 use super::{BlobStore, NodeId, Result};
 
+#[derive(Debug, Clone)]
 pub struct MemStore {
```

Comment:
Not for this PR, but if you need cheap clone we should probably impl the mem store on top of immutable/persistent underlying data structure.

There are two major crates I am aware of that do this. One is https://docs.rs/im/latest/im/ , but looking at github this seems abandoned. The other is https://docs.rs/rpds/latest/rpds/, which is still active.

https://github.com/orium/rpds

Reply:
Made an issue for this: https://github.com/n0-computer/willow-store/issues/5